src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Extensions/RangeExtensions.cs

```diff
@@ -44,5 +42,112 @@ public static bool IsUndefined(this Range range)
 
             return range == UndefinedRange;
         }
+
+        public static bool OverlapsWith(this Range range, Range other)
```

Comment:
We had two Range extension classes in the `LangaugeServerClient.Razor` project so I combined them into one.